pgjdbc/src/main/java/org/postgresql/jdbc/PgStatement.java

```diff
@@ -355,6 +355,11 @@ protected void closeForNextExecution() throws SQLException {
     // Close any existing resultsets associated with this statement.
     synchronized (this) {
       closeUnclosedResults();
+
+      /* we are going to release this anyway make sure it is closed */
+      if ( this.result != null && this.result.getResultSet() != null ) {
+        ((PgResultSet)this.result.getResultSet()).closeInternally();
+      }
```

Comment:
Should this be placed in `closeUnclosedResults`?
Otherwise, it sounds strange that `closeUnclosedResults` leaves an unclosed resultset behind.

Reply:
probably yes, one more Q should we include this in 42.2.19 ?

Reply:
Let's include it provided it passes tests.